File: examples/Fortran/nd.f90
! examples/Fortran/nd.f90 - Example code for SPRAL_ND package
program example
   use spral_nd
   implicit none

   ! Derived types
   type (nd_options) :: options
   type (nd_inform) :: inform

   ! Matrix data
   integer :: n, row(14), ptr(9), perm(8)

   ! Data for matrix:
   !     1  2  3  4  5  6  7  8
   ! 1 (    X     X  X  X       )
   ! 2 ( X        X     X  X    )
   ! 3 (             X  X       )
   ! 4 ( X  X              X    )
   ! 5 ( X     X        X     X )
   ! 6 ( X  X  X     X     X  X )
   ! 7 (    X     X     X       )
   ! 8 (             X  X       )
   n = 8 
   ptr(1:n+1)        = (/ 1,          5,       8,   10, 11,  13,   15, 15, 15 /)
   row(1:ptr(n+1)-1) = (/ 2, 4, 5, 6, 4, 6, 7, 5, 6, 7, 6, 8, 7, 8 /)

   ! Find nested dissection ordering
   call nd_order(0, n, ptr, row, perm, options, inform)

   ! Print out nested dissection ordering
   write(*,"(a, 8i8)") ' Permutation : ', perm(:)
end program example

File: examples/C/nd.c
/* examples/Fortran/nd.f90 - Example code for SPRAL_ND package */
#include "spral.h"
#include <stdio.h>

int main(void) {
   /* Derived types */
   struct spral_nd_options options;
   struct spral_nd_inform inform;

   /* Initialize options */
   spral_nd_default_options(&options);

   /* Data for matrix:
    *     0  1  2  3  4  5  6  7
    * 0 (    X     X  X  X       )
    * 1 ( X        X     X  X    )
    * 2 (             X  X       )
    * 3 ( X  X              X    )
    * 4 ( X     X        X     X )
    * 5 ( X  X  X     X     X  X )
    * 6 (    X     X     X       )
    * 7 (             X  X       ) */
   const int n = 8;
   int ptr[] = { 0,          4,       7,    9, 10,  12,   14, 14, 14 };
   int row[] = { 1, 3, 4, 5, 3, 5, 6, 4, 5, 6, 5, 7, 6, 7 };

   /* Find and print nested dissection ordering */
   int perm[n];
   spral_nd_order(0, n, ptr, row, perm, &options, &inform);
   printf("Permutation:");
   for(int i=0; i<n; i++) printf(" %5d", perm[i]);
   printf("\n");

   return 0;
}
